### DISCUSSION TITLE
implement math.rank

### DISCUSSION THREAD

**Discussion by ORESoftware** [josdejong/mathjs#2798] [Ideas]
Not a bug, a feature request:

Doesn't look like I can easily find the rank of a matrix. (Maybe I can do it in two steps given existing methods?). Would like a function call like:

```js
math.rank( [ [1,0], [2,1] ] );

```

since row rank = col rank, I think one call should be sufficient...I suppose people could always transpose and re-call math.rank if they really wanted to?

**Comment by josdejong** [comment]
Thanks for your suggestion, implementing `math.rank` would be very useful indeed.

Anyone interested in implementing this? Help would be welcome

**Comment by gwhitney** [comment]
Warning: computing the rank of a matrix of floating point numbers is numerically intricate, see for example: https://math.stackexchange.com/questions/2238798/numerically-determining-rank-of-a-matrix. So I think it usually uses the singular value decomposition of a matrix, which does not seem to be implemented in mathjs yet.
